### PR TITLE
Export scanner no longer shows shipping manifest value + manifest error changes

### PR DESCRIPTION
**Pull request by E231kei500bandai**
## About The Pull Request

Adds a variable to exports called `scannable`. If you are trying to scan it and its set to false, then it won't show the value. When the actual sale is done, the value will still be added/subtracted.

I also reorganized the message the scanner sends because I didn't want to put multiple inline if statements.

Lastly, made it so that locked crates and private order lockboxes can't get the shipping manifest contents error by adding a new trait `TRAIT_NO_MANIFEST_CONTENTS_ERROR`. For departmental orders, I just entirely disabled the manifests having errors by setting it to `MANIFEST_CAN_FAIL` to false. 

This PR conflicts with #78911, sorry but that PR was what reminded me to fix this bug.

## Why It's Good For The Game

Some may call it tedium but having to read paperwork is part of the cargo experience. Since it is harder to determine the crates value, there has been a few balance changes. The penalty for incorrect manifest is half the crates price or 500 credits, whichever is less.

image

Let me know if you have any suggestions for how the message is worded, it might be too long. 

Locked crates and private orders can no longer get the shipping manifest contents error because it is basically impossible to get someone to tell you all the contents of their purchase. It's only a 5% chance of being wrong and if they're not in cargo they probably don't care about the cargo budget.

## Changelog
:cl:
balance: Export scanner no longer shows value of shipping manifests, now you actually have to read them.
balance: Shipping manifest penalty is now only half crate cost as well as capped to 500 credits. 
balance: Shipping manifests for private orders or locked crates can no longer have the incorrect contents error. Shipping manifests for departmental orders can n longer have any error.
/:cl:

### CODE REVIEW

**Comment by GPeckman** [comment]
> Some may call it tedium but having to read paperwork is part of the cargo experience.

There are actually some edge cases where paperwork appears incorrect but is in reality correct. If the captain renames the station, for instance, the manifest will appear to have the wrong station name even though its actually right.

There are also cases like private orders and mining/bitrunner orders. The requisition form doesn't say what was bought, only who bought it, so if you need to verify if the manifest is correct then you have to interrogate the person who made the order.

**Comment by BlueMemesauce** [comment]
well i guess i have to fix those bugs now

**Comment by mc-oofert** [comment]
if anything this should be a trait

**Comment by BlueMemesauce** [comment]
> There are actually some edge cases where paperwork appears incorrect but is in reality correct. If the captain renames the station, for instance, the manifest will appear to have the wrong station name even though its actually right.

Fixed, it can only occur after the station name is set in stone (5 mins after the round starts)

> There are also cases like private orders and mining/bitrunner orders. The requisition form doesn't say what was bought, only who bought it, so if you need to verify if the manifest is correct then you have to interrogate the person who made the order.

Why would you need to do this? The only errors that can occur are removing things from the list. If it doesn't say what was bought, just deny it because an empty manifest is wrong.

**Comment by RedBaronFlyer** [comment]
I apologize in advance for posting an essay:

To preface, I love cargo, so I am massively biased, but manifests are in this weird spot. Some stuff is pretty easy to check:

- Crates listing an incorrect number in a shipment (laser crate listing it only having two lasers (laser crates always have three)),
- Duplicates (so a plastic order listing plastic twice)
- The station name is incorrect (this can suck if your station name is a generic one and the incorrect station name is also a generic one)
- The manifest is empty.
- some other system I am surely missing (seriously, I have nearly two years of cargo experience, and I still don't get why the manifest is wrong sometimes)

The big issue is verifying the contents of the crate or access restricted crates because it can be hard to tell:
This usually rears its head with things like donk-variety crates, medical supplies, blood supplies (I think?), etc. Anything with a randomized pool of items in the crate can quickly get messy.

Anything that cargo itself can open itself is _usually_ pretty easy. Still, some orders can have incorrect contents and also be access restricted, which sucks. Crates you can open but have a **MASSIVE** pool of items can also suck. You don't know pain until you've experienced manually checking a medical supply crate to see if the 20 items in it match the 20 items listed on the manifest. I've done it and wanted to jump out of an airlock because searching through stacks sucks in SS13. Now imagine doing that for five medical supply crates on one order (this has also happened to me)

There is already a massive problem of people needing to return crates, and people usually only bother to yell at cargo if an order is empty. I get trying to double-check contents, but sometimes the interactions feel like this when you deliver a crate and its access restricted:

I have done the prior before, said conversations go like this:
"hey, is this in there?"
"yes"
"is this in there?"
"yes"
"is that in there?"
"yep"
"is that thing in there?"
"yeah"
"are there two doodads in there?"
"yes."
"how about this thing?"
_they walk away_

Part of me is worried that if this goes through, it will simply encourage people not to even bother with manifest because getting 80 credits back is not worth hundreds or thousands of credits being deducted. I get the idea of the manifest thing, and I like the concept. In execution, it's mostly a noob trap for when a new cargo tech is left dumbfounded when they stamped a BZ manifest for "delta delta bravo whisky 7 station" instead of "delta whisky delta bravo 7 station," and it instantly evaporates 9k of cargo's budget with the vague message that an (unspecified, mind you) manifest was erroneous. (which they might not even see if the shuttle is constantly moving.)

I think that people will continue to do paperwork, but at the same time, running the risk of deducting massive amounts of credits (as the fine is the full cost of the order, if I recall right) will make it really iffy. getting back 80 credits may just not seem worth the headache since 80 credits can add up, but it's also minor enough that you could just ignore it. It would only be worth it for super obvious things like a blank manifest, for instance.

This isn't even to mention the issue of other cargo techs ripping off the manifest when whisking the order away to whoever ordered it. It's not uncommon to walk onto the cargo shuttle or into the cargo bay and see a pile of manifest lying on the ground as if the cargo crate version of the rapture happened.

I don't like the current system of gaming manifest by just mass stamping them and export scanning to see which ones ring negative, but at the same time, sometimes checking contents feels out of your control, which sucks. Private orders are a prominent example of this. Sometimes, I've gotten dinged because the person ordered two mugs, the order had one, didn't say anything, and then cargo got fined for approving an erroneous manifest. I don't know if private orders can still be incorrect regarding their contents, so that that bit could be outdated.

**Comment by RedBaronFlyer** [comment]
Guillaume brought up an idea on the Manuel Discord about some sort of goggles to check the contents of crates and lockers to check the contents within. I had half-heartedly joked about an airport-style luggage checker machine that you would use to check the contents of the crate and it'd show the contents of said crate on a TGUI screen so you could quickly visually compare it to a manifest without having to manually open it and sort through god knows how many items. Meanwhile, you would still need to check stuff like the station name, empty manifest, and duplicate item listings (plastic crate manifest listing plastic twice, for instance.)

I realize those would be massively out of the scope of this PR, though.

**Comment by FlufflesTheDog** [comment]
[RedBaronFlyer](https://github.com/RedBaronFlyer) brought up pretty much the exact same points I wanted to. I'm absolutely one of those people who won't bother approving the majority of manifests since there's so many where you can get got by something silly, and the reward is so tiny compared to the potential for thousands of credits lost, plus the amount of effort it takes for the (on average) tiny gain.

I'm not sure if I'd call this a good solution, but something that comes to mind is just making certain manifest errors not punish cargo. You could argue that CC wouldn't care to (or even know to) fine you for approving something that didn't have everything it was supposed to, for example.

**Comment by BlueMemesauce** [comment]
> [RedBaronFlyer](https://github.com/RedBaronFlyer): essay

I get what you mean now, if the manifest isn't totally empty but something was still removed, it's impossible to tell without asking the person who made the order. I removed this type of error from private order manifests. the only error they can get is incorrect station name. 

I reduced the the penalty for incorrect manifest. It's half the crates price or 500 credits, whichever is less. I also reduced the reward to half the crates price.

**Comment by YuliceTheDruid** [comment]
IF the reward for incorrect manifests is reduced to half then that means that whenever the shipping order is wrong in the sense that the object didn't show up, then you now are losing money by ordering.

**Comment by Jacquerel** [comment]
I am sympathetic to the goals here but ultimately I think if you don't want there to be a shortcut to checking the contents of a crate matches the manifest people will simply encourage each other to never stamp manifests at all, the risk isn't worth the reward and checking the contents of crates you can't open is a kind of tedium which creates significant risk.
It might make more sense to just canonise this, add some other form of being able to scan crates to check if their contents are as expected, or remove that from consideration (although it would be sad to lose out on missing deliveries and it would be sad not to be able to get a refund when it happens).

**Comment by Jacquerel** [comment]
> I don't know, I think the the current system seems more interesting. You have to actually put some effort in to notice wrong paperwork and there's risk and reward. I don't think stamping and then scanning all the manifests is particularly interesting. Or any other machine that just tells you if it's right or not without you having to use any mental effort.

If the changes have made it so that you don't need to manually count what's inside a crate to verify if it was correct I don't have an issue, but it is that behaviour which is problematic as often cargo workers won't be present (or sometimes even _allowed_ to be present) when a crate is opened and have no means of verifying it before then.
A scanner just to check the _contents_ is what I'd like, rather than one which just says "something on the manifest is incorrect".

**Comment by BlueMemesauce** [comment]
You mean when people take the crates from the cargo shuttle before any cargo techs had a chance to check the manifest? That seems like an uncommon occurrence that can be prevented. Just check the crates on the shuttle before bringing it into the cargo bay if there's a ton of assistants waiting there. If they're charging the shuttle the moment it arrives, maybe call security to tell the tiders to go away.

**Comment by YuliceTheDruid** [comment]
> You mean when people take the crates from the cargo shuttle before any cargo techs had a chance to check the manifest? That seems like an uncommon occurrence that can be prevented. Just check the crates on the shuttle before bringing it into the cargo bay if there's a ton of assistants waiting there. If they're charging the shuttle the moment it arrives, maybe call security to tell the tiders to go away.

More like crates that have locks and the contents can be erroneous, IE missing or the paper is wrong. Sometimes the paper will have duplicates listed on the sheet or be missing an item entirely, or have items that aren't on the manifest listed in it which you'd have to check. Along with the name being incorrect. Again, though, in the case of items missing if you cut the reward from correctly stamped wrong manifests to half then crates that don't have the contents and are stamped wrong are still causing you to lose 50% of your money.

**Comment by BlueMemesauce** [comment]
> More like crates that have locks and the contents can be erroneous, IE missing or the paper is wrong. Sometimes the paper will have duplicates listed on the sheet or be missing an item entirely, or have items that aren't on the manifest listed in it which you'd have to check. Along with the name being incorrect. Again, though, in the case of items missing if you cut the reward from correctly stamped wrong manifests to half then crates that don't have the contents and are stamped wrong are still causing you to lose 50% of your money.

Private orders and locked crates already couldn't have missing items in the crate. I changed it so they can't get wrong manifest content either so these orders can now only get wrong station name errors.

I also just changed the reward for correctly denied manifests to refund the full value.

**Comment by EricZilla** [comment]
You know I used to think empty crates was just a glitch that happened when you ordered stacks of materials because it literally somehow ONLY happened when I ordered a few crates of iron to help the matconomy, because there is literally nothing anywhere in game to suggest this could be an intentional design or how to fix it, I only finally found out this was a game mechanic because I bought a dog and he was no where to be found.

Basically I'm saying this is just kind of shit, but thats out of scope here. If you are going to make it so people have to manually read manifests, you should increase the reward for them being handled correctly not decrease it. If this gets implemented I'm just burning all manifests for boxes cargo cant open.

File: code/modules/cargo/universal_scanner.dm [hunk 1]
```diff
@@ -173,10 +173,31 @@
 	var/price = 0
 	for(var/exported_datum in report.total_amount)
 		price += report.total_value[exported_datum]
-	if(price)
-		to_chat(user, span_notice("Scanned [target], value: <b>[price]</b> credits[target.contents.len ? " (contents included)" : ""]."))
+
+	var/message = "Scanned [target]"
+	var/warning = FALSE
+	if(target.contents.len)
```
Comment (Jacquerel) [inline code/modules/cargo/universal_scanner.dm:179]:
```suggestion
	if(length(target.contents))
```
Minor note but generally prefer `length()` to `.len`, mostly for null handling.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/__DEFINES/station.dm [hunk 2]
```diff
@@ -14,3 +14,6 @@
 
 /// The data file that future station traits forced by an admin are stored in
 #define FUTURE_STATION_TRAITS_FILE "data/future_station_traits.json"
+
+/// The amount of time until the station charter can no longer be used to rename the station
+#define STATION_RENAME_TIME_LIMIT 3000
```
Comment (Ghommie) [inline code/__DEFINES/station.dm:19]:
```suggestion
#define STATION_RENAME_TIME_LIMIT 5 MINUTES
```

File: code/modules/cargo/order.dm [hunk 3]
```diff
@@ -136,11 +136,15 @@
...
 	if((manifest_paper.errors & MANIFEST_ERROR_CONTENTS) && container_contents)
-		for(var/i = 1 to rand(1, round(container.contents.len * 0.5))) // Remove anywhere from one to half of the items
-			var/missing_item = pick(container_contents)
-			container_contents[missing_item]--
-			if(container_contents[missing_item] == 0) // To avoid 0s and negative values on the manifest
-				container_contents -= missing_item
+		if(HAS_TRAIT(container, TRAIT_NO_MANIFEST_CONTENTS_ERROR))
+			manifest_paper.errors &= ~MANIFEST_ERROR_CONTENTS
+		else
+			for(var/i = 1 to rand(1, round(container.contents.len * 0.5))) // Remove anywhere from one to half of the items
```
Comment (Ghommie) [inline code/modules/cargo/order.dm:142]:
```suggestion
			for(var/iteration in1 to rand(1, round(container.contents.len * 0.5))) // Remove anywhere from one to half of the items
```

**Review by Ghommie — CHANGES_REQUESTED** [review]
